crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -364,8 +369,9 @@ pub struct Backend64Bit<
     phantom_asm: PhantomData<ASM>,
     phantom_cc: PhantomData<CC>,
     target_info: TargetInfo,
-    env: &'a Env<'a>,
-    interns: &'a mut Interns,
+    env: &'r Env<'a>,
+    layout_interner: &'r mut STLayoutInterner<'a>,
+    interns: &'r mut Interns,
```

Comment:
All this lifetime stuff is frustrating, but it is necessary because we do care about borrowing env/layout interner/interns disjointly from the lifetime of the arena they reference (the arena lives longer usually)